### PR TITLE
CI-install

### PR DESCRIPTION
**Pull request by earny-joe**
Looking to merge updates made in explore branch to master. Changes include:

- added `pytest`, `pytest-cov` and `flake8` to environment
- updated requirements.txt file to include all necessary libraries
- added `.travis.yml`, `.coveragerc` files
- updated `setup.py` file

Additionally, I starting cleaned up the `experiments` folder and put more experimental-focused work in `playground_nbs` subfolder.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: python
+python:
+ - '3.6'
+install:
+ - pip install -r requirements.txt
+script:
+ - pytest --suppress-no-test-exit-code
```
Comment (rkingery) [inline .travis.yml:7]: Make sure this suppress flag is only temporary. Don't want this in the whole time.

File: docs/conf.py [hunk 2]
```diff
@@ -3,28 +3,28 @@
...
 #
 # Note that not all possible configuration values are present in this
 # autogenerated file.
 #
 # All configuration values have a default; values that are commented out
 # serve to show the default.
 
-import os
-import sys
+# import os (original imported these, however flake8 said they were unused)
```
Comment (rkingery) [inline docs/conf.py:14]: If you're not using, always remove them. Don't comment them out unless you *know for a fact* you're going to use them *soon*.

File: docs/conf.py [hunk 3]
```diff
@@ -3,28 +3,28 @@
...
+# import sys (original imported these, however flake8 said they were unused)
 
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 # sys.path.insert(0, os.path.abspath('.'))
```
Comment (rkingery) [inline docs/conf.py:20]: Find a better way to deal with paths than this. I recommend always using *relative imports* in your project for imports to your own code. If you want to treat them as absolute paths instead, you need to properly pip install your project as a package.

File: experiments/5.0-je-v1-streamlistener.ipynb [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.7.6"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 4
+}
```
Comment (rkingery) [inline experiments/5.0-je-v1-streamlistener.ipynb:389]: Why do you have both an ipynb and a py version of this? Is it essential that both of these be visible in the project?

File: experiments/5.0-je-v1-streamlistener.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            try:
+                csvwriter.writerow(
+                    [
+                        status.created_at, status.id_str, status.user.id_str,
+                        status.user.screen_name, status.user.name, status.text,
+                        status.reply_count, status.retweet_count,
+                        status.favorite_count
+                    ]
+                )
+            except Exception as e:
```
Comment (rkingery) [inline experiments/5.0-je-v1-streamlistener.py:35]: As a general rule, *never* write general exceptions. Always try to catch only the exact type of error that you're trying to catch, e.g. `ValueError` or `IndexError` or whatever.

File: experiments/5.0-je-v1-streamlistener.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+                print(e)
+                pass
+        csvfile.close()
+        return
+
+    def on_error(self, status_code):
+        print("Encountered error with status code:", status_code)
```
Comment (rkingery) [inline experiments/5.0-je-v1-streamlistener.py:42]: From now on, use only python format strings for printing. This one would be written instead as:

```
print(f"Encountered error with status code: {status_code}")
```

Also, minor note: Python style prefers the single quote over the double quote by default. Most important thing here is to be consistent, but just be aware. Some people are more picky on this stuff than I am.

File: experiments/fundamentals.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import os
+import pandas as pd
+
+
+def load_data(origpath, datapath, filename):
+    """
+    Given path to a specific data directory, loads in data from given filename
+    """
+    # change directory to where data is located
+    os.chdir(datapath)
```
Comment (rkingery) [inline experiments/fundamentals.py:10]: Try to start using the pathlib functionality. Try to get out of using os for dealing with paths.

**Comment by earny-joe** [comment]
Made changes to the files mentioned above and removed files/folders that were unnecessary in the most recent commit.

**Pull request merged**